ericscript_to_bedpe.py

```diff
@@ -0,0 +1,117 @@
+
+'''
+Convert EricScript output to BEDPE format.
+
+Can be run from the commandline with:
+    python fusioncatcher_to_bedpe.py -i <input_file> -o <output_file>
+
+If no <input_file> and/or <output_file> is specified, the file will read/write
+from/to STDIN/STDOUT, allowing for piping into and out of the program.
+'''
+
+import argparse,sys,csv
+
+def map_fields(input_row, headings):
+    '''
+    Map fields from the input row to bedpe_fields
+
+    Args:
+        input_row (dict): {'heading': value} mapping for one row in input
+        headings (list): A list of all BEDPE headings to include
+    Returns:
+        dict: the return value. A single BEDPE row
+    '''
+    out_row = dict()
+
+    [chr1, pos1, strand1 ] = input_row['chr1'], input_row['Breakpoint1'], input_row['strand1']
+
+    out_row['chrom1'] = 'chr{num}'.format(num=chr1)
+    if(pos1.isdigit()):
+        out_row['start1'] = int(pos1) - 1
+        out_row['end1'] = int(pos1)
+    else:
+        out_row['start1'] = -1
+        out_row['end1'] = -1
+
+    out_row['strand1'] = strand1
+
+    [chr2, pos2, strand2 ] = input_row['chr2'], input_row['Breakpoint2'], input_row['strand2']
+
+    out_row['chrom2'] = 'chr{num}'.format(num=chr2)
+
+    if(pos2.isdigit()):
+        out_row['start2'] = int(pos2) - 1
+        out_row['end2'] = int(pos2)
+    else:
+        out_row['start2'] = -1
+        out_row['end2'] = -1
+
+    out_row['strand2'] = strand1
+
+
+    gene1 = input_row['GeneName1']
+    gene2 = input_row['GeneName2']
+    out_row['name'] = '{G1}-{G2}'.format(G1=gene1, G2=gene2)
+
+    out_row['score'] = input_row['EricScore']
```

Comment:
Note that EricScore is an aggregate  of other scores employed by the author, which are reported in the output. See paper: [](https://academic.oup.com/bioinformatics/article/28/24/3232/247436/Discovering-chimeric-transcripts-in-paired-end-RNA)

Reply:
@rdocking any suggestion?

Reply:
No strong preference here